core/src/Cabana_LinkedCellList.hpp

```diff
@@ -431,15 +566,128 @@ void permute(
  */
 template <class LinkedCellListType, class SliceType>
 void permute(
-    const LinkedCellListType& linked_cell_list, SliceType& slice,
+    LinkedCellListType& linked_cell_list, SliceType& slice,
     typename std::enable_if<( is_linked_cell_list<LinkedCellListType>::value &&
                               is_slice<SliceType>::value ),
                             int>::type* = 0 )
 {
     permute( linked_cell_list.binningData(), slice );
+
+    // Update internal state.
+    linked_cell_list.update( true );
 }
 
 //---------------------------------------------------------------------------//
+//! Stencil of cells surrounding each cell.
+/*
+template <class Scalar>
+struct LinkedCellStencil
```

Comment:
I would rather keep this, then add an interface `LinkedCellList::getStencil` which calls `getCells`